sdk/communication/azure-communication-jobrouter/src/main/java/com/azure/communication/jobrouter/models/JobMatchingMode.java

```diff
@@ -28,11 +29,24 @@
     @JsonSubTypes.Type(name = "suspend", value = SuspendMode.class) })
 @Immutable
 public class JobMatchingMode {
```

Comment:
@cparisineti You need to rename this JobMatchingModeInternal like all the other base classes.  Then you are free to add the Kind property and make it abstract.